src/transaction/Transaction.js

```diff
@@ -868,9 +837,35 @@ export default class Transaction extends Executable {
                 signedTransaction.sigMap.sigPair = [];
             }
 
-            signedTransaction.sigMap.sigPair.push(
-                publicKey._toProtobufSignature(signatureArray[index]),
-            );
+            if (signedTransaction.bodyBytes) {
+                const { transactionID, nodeAccountID } =
+                    HashgraphProto.proto.TransactionBody.decode(
+                        signedTransaction.bodyBytes,
+                    );
+
+                if (!transactionID || !nodeAccountID) {
+                    throw new Error(
+                        "Transaction ID or Node Account ID not found in the signed transaction",
+                    );
+                }
+
+                const transactionId =
+                    TransactionId._fromProtobuf(transactionID);
+                const nodeAccountId = AccountId._fromProtobuf(nodeAccountID);
+
+                const signature = signatureMap
```

Comment:
IMO: You can split this nesting into 3 different variables that way it will be more readable.

```
const accountSignatures / nodeSignatures = signatureMap.get(nodeAccountId);
const transactionSignatures = accountSignatures?.get(transactionId);
const signature = transactionSignatures?.get(publicKey);
```